### PR TITLE
feat(api/v2): implement TOTP session check

### LINKED ISSUE #5477: API: Login with MFA (TOTP/U2F)
As a developer I want to be able to login a user with factors like otp and u2f through the API, so I can build my own Login UI.

**Acceptance Criteria**
- [x] Extend session checks with OTP
- [x] I am able to start webauthn check which returns a challenge
- [x] I am able to specify with which platform the authentication should happen (platform, cross-plattform, etc)
- [x] Extend session checks with WebAuthN
- [x] Session is enriched with the factor otp or webauthn if i have used it to authenticate
- [x] Follow up TODOs:

https://github.com/zitadel/zitadel/blob/8dc1fd06a1068e162db6c2986e402ff349a3964c/internal/command/session_model.go#L179-L187

https://github.com/zitadel/zitadel/blob/041d68824ea6c79ec4211665aa2818f1af649b7d/internal/command/session_model.go#L228-L233

### PR DESCRIPTION
**Pull request by muhlemmer**
This change implements Time-based One-Time Password (TOTP) on the v2alpha session API.

Closes #5477 

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/repository/session/session.go [hunk 1]
```diff
@@ -264,6 +265,39 @@ func NewWebAuthNCheckedEvent(
...
+func NewTOTPCheckedEvent(
+	ctx context.Context,
+	aggregate *eventstore.Aggregate,
+	checkedAt time.Time,
+) *TOTPCheckedEvent {
+	return &TOTPCheckedEvent{
+		BaseEvent: *eventstore.NewBaseEventForPush(
+			ctx,
+			aggregate,
+			PasswordCheckedType,
```
Comment (livio-a) [inline internal/repository/session/session.go:295]:
```suggestion
			TOTPCheckedType,
```

File: internal/query/projection/session.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ const (
 	SessionColumnIntentCheckedAt      = "intent_checked_at"
 	SessionColumnWebAuthNCheckedAt    = "webauthn_checked_at"
 	SessionColumnWebAuthNUserVerified = "webauthn_user_verified"
+	SessionColumnTOTPCheckedAt        = "totp_checked_at"
```
Comment (livio-a) [inline internal/query/projection/session.go:33]: please increase the projection version, so the column will get created on existing databases

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/query/projection/session.go (hunk 2 above)
Comment (muhlemmer) [reply internal/query/projection/session.go:33]: 3 to 4 was done already in https://github.com/zitadel/zitadel/pull/6339 as part of this sprint.

**Review by livio-a — APPROVED** [review]

**Comment by dimyle** [comment]
Hello. My version is 2.35. I've just tried TOTP + password. I've added TOTP and created session with 3 checks (username, password, totp). When i request /auth/v1/users/me, I get an error "mfa required (AUTHZ-Kl3p0)". Unfortunately I'm not a GO developer, I can't find an exact problem on myself. But it looks that a problem is in file "token_verifier.go" line 203, there is a comment left, pointing to head issue of this task. Can someone confirm please?

**Comment by hifabienne** [comment]
> 

Hei @dimyle 
It looks like on the organisation mfa is required, in that case your user needs to authenticate with a factor and not only password.

**Comment by dimyle** [comment]
> > 
> 
> Hei @dimyle It looks like on the organisation mfa is required, in that case your user needs to authenticate with a factor and not only password.

Hello @hifabienne!
That's true, organization MFA is required. In my tests it was set up and working. I've created session with 3 checks (login/password/TOTP) - that is working good. But, when I try to get user info, as i mentioned in my message above, this call fails. There is a commented piece of code in file, that prevent normal work of TOTP. If I use Email as MFA (not TOTP), everything works good.

**Comment by muhlemmer** [comment]
can you please raise an issue, so we can track and solve it. It's hard to keep track of closed PRs. Also, can you include a [permalink](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet) to the suspected code, it makes it easier for everybody.

I will try to pick it up this Friday. (our bugfix day).

Thanks.

**Pull request merged**